compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -3118,9 +3118,12 @@ private STNode parseStatement(STNode annots) {
                 // any statement starts with `[` can be either a var-decl with tuple type
                 // or a destructuring assignment with list-binding-pattern.
                 return parseStatementStartsWithOpenBracket(getAnnotations(annots), false);
+            case TRANSACTIONAL_KEYWORD:
+                if (peek(2).kind == SyntaxKind.WORKER_KEYWORD) {
+                    return parseNamedWorkerDeclaration(getAnnotations(annots));
+                }
```

Comment:
Do we need recovery for worker keyword @SupunS  WDYT?

Reply:
yeah, I think we need to. 
Otherwise if there are additional tokens like:
```ballerina
transactional someJunkToken worker w1 {

}
```
or if the worker keyword is missing like:

```ballerina
transactional  w1 {

}
```
the parsing will go haywire.

Reply:
Shall I open another issue to add recovery for worker keyword and link it to this PR? @SupunS @rdulmina

Reply:
+1

Reply:
+1. But lets fix it soon, since this is a blocker for auto-completion in LS.

Reply:
We will be fixing this in a different issue https://github.com/ballerina-platform/ballerina-lang/issues/26176